mqtt-connect/model/src/main/java/com/cumulocity/mqtt/connect/client/model/MqttMessage.java

```diff
@@ -0,0 +1,33 @@
+package com.cumulocity.mqtt.connect.client.model;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+/**
+ * MQTT message representation containing a payload and associated {@link MqttMetadata metadata}.
+ */
+@NoArgsConstructor
+@AllArgsConstructor
+@Data
+public class MqttMessage {
+
+    private byte[] payload;
+
+    private MqttMetadata metadata;
+
+    /**
+     * @return the payload of this message as a byte array. This will be an empty array if the payload is not present.
+     */
+    public byte[] getPayload() {
```

Comment:
question: do we need the getters given the @Data annotation is used? Same for other methods and MqttMetadata

Reply:
They are just for proper javadoc generation, otherwise generated javadoc doesn't any doc :)

Reply:
Thanks